src/Functions/getMergeTreeSetting.cpp

```diff
@@ -0,0 +1,118 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <DataTypes/FieldToDataType.h>
+#include <Interpreters/convertFieldToType.h>
+#include <Interpreters/Context.h>
+#include <Core/Field.h>
+#include <Storages/MergeTree/MergeTreeSettings.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+class FunctionGetMergeTreeSetting : public IFunction, WithContext
+{
+public:
+    static constexpr auto name = "getMergeTreeSetting";
+
+    static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionGetMergeTreeSetting>(context_); }
+    explicit FunctionGetMergeTreeSetting(ContextPtr context_) : WithContext(context_) {}
+
+    String getName() const override { return name; }
+
+    bool isDeterministic() const override { return false; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    size_t getNumberOfArguments() const override { return 1 ; }
+
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0}; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (arguments.size() != 1)
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Number of arguments for function {} can't be {}, should be 1",
+                getName(),
+                arguments.size());
+
+        if (!isString(arguments[0].type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "The argument of function {} should be a constant string with the name of a setting",
+                String{name});
+
+        const auto * column = arguments[0].column.get();
+        if (!column || !checkAndGetColumnConstStringOrFixedString(column))
+            throw Exception(
+                ErrorCodes::ILLEGAL_COLUMN,
+                "The argument of function {} should be a constant string with the name of a setting",
+                String{name});
+
+        std::string_view setting_name{column->getDataAt(0).toView()};
+
+        auto value = getContext()->getMergeTreeSettings().get(setting_name);
+
+        return applyVisitor(FieldToDataType{}, value);
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
+    {
+        if (arguments.size() != 1)
+            throw Exception(
```

Comment:
Why are we doing `if (arguments.size() != 1)` in two places - once in `getReturnTypeImpl` L42 and again here? Ideally we should only do this in `getReturnTypeImpl`.

Reply:
nvm, I guess this is similar to `getSetting` function and since we don't know the exact type of setting that will be returned, I guess, it needs validation in both places. I didn't notice it earlier. Also, optional, but let's add back the private helper function if you think if it makes it cleaner: https://github.com/ClickHouse/ClickHouse/pull/78439#discussion_r2027912320. Thank you!

Reply:
nvm, I guess wee need it in both `executeImpl` and `getReturnTypeImpl` for this particular function - similar to `getSettings` function. Didn't notice it ealier.